### PR TITLE
chore: Rename VerifiableCredentialsFactory to JwtUtils

### LINKED ISSUE #1714: DID: VerifiableCredentialFactory class name
# Feature Request

The class  has the doc *Convenience/helper class to generate, verify and deserialize verifiable credentials, which are, in fact, Signed JSON Web Tokens (JWTs).*

In fact the implementation is only based on DID and JWT, and does not contain any claims from the verifiable credentials data model. It is used for generating and validating short-lived authentication tokens in synchronous HTTP requests - and not for managing verifiable credentials, which will be done in other components (IdentityHub, MVD).

For clarity the class should be renamed to something like `DidJwtUtils`. The methods could be made non-static for better testability of dependent classes.

Trying to extend the class to also support the verifiable credential data model wouldn't seem to be a good idea, since there are specifics to usage as a web request token (audience, expiration, etc.)

## Which Areas Would Be Affected?
Identity DID

## Type of Issue
improvement

## Checklist

- [ ] assigned appropriate label?
- [x] **Do NOT select a milestone or an assignee!**

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

Rename `VerifiableCredentialsFactory` to `JwtUtils` and move it to identity-did-service (single place where it's used)

## Why it does that

The class had the doc `Convenience/helper class to generate, verify and deserialize verifiable credentials, which are, in fact, Signed JSON Web Tokens (JWTs).`

In fact the implementation is only based on DID and JWT, and does not contain any claims from the verifiable credentials data model. It is used for generating and validating short-lived authentication tokens in synchronous HTTP requests - and not for managing verifiable credentials, which will be done in other components (IdentityHub, MVD).

Trying to extend the class to also support the verifiable credential data model wouldn't seem to be a good idea, since there are specifics to usage as a web request token (audience, expiration, etc.)

## Linked Issue(s)

Closes #1714

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java [hunk 1]
```diff
@@ -59,7 +58,7 @@ public DecentralizedIdentityService(DidResolverRegistry resolverRegistry, Creden
 
     @Override
     public Result<TokenRepresentation> obtainClientCredentials(TokenParameters parameters) {
-        var jwt = VerifiableCredentialFactory.create(privateKey, issuer, parameters.getAudience(), clock);
+        var jwt = JwtUtils.create(privateKey, issuer, issuer, parameters.getAudience(), clock);
```
Comment (paullatzelsperger) [inline extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java:61]: micro nit: static import?

File: extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java [hunk 2]
```diff
@@ -88,7 +87,7 @@ public Result<ClaimToken> verifyJwtToken(TokenRepresentation tokenRepresentation
             PublicKeyWrapper publicKeyWrapper = KeyConverter.toPublicKeyWrapper(publicKeyJwk, publicKey.get().getId());
 
             monitor.debug("Verifying JWT with public key...");
-            Result<Void> verified = VerifiableCredentialFactory.verify(jwt, publicKeyWrapper, audience);
+            Result<Void> verified = JwtUtils.verify(jwt, publicKeyWrapper, audience);
```
Comment (paullatzelsperger) [inline extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java:90]: micro nit: static import?

File: extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java (hunk 1 above)
Comment (algattik) [reply extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java:61]: Are you sure? It seems the version without static import is clearer

File: extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java (hunk 2 above)
Comment (algattik) [reply extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java:90]: As above - Are you sure? It seems the version without static import is clearer

File: extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java (hunk 1 above)
Comment (paullatzelsperger) [reply extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java:58]: for me, reading `var jwt = create(...)` leaves little room for interpretation :)

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**